python/cucim/src/cucim/skimage/measure/_regionprops_gpu_misc_kernels.py

```diff
@@ -0,0 +1,721 @@
+        possible error that would otherwise occur in this case. Turning this
+        on will make the run time substantially longer, so it should only be
+        used when labeled regions may have a non-negligible portion of their
+        boundary within a <2 pixel gap from another label.
+    labels_close : numpy.ndarray or sequence of int
+        List of labeled regions that are less than 2 pixel gap from another
+        label. Used when robust=True. If not provided and robust=True, it
+        will be computed internally.
+    props_dict : dict or None, optional
+        Dictionary of pre-computed properties (e.g. "slice"). The output of this
+        function will be stored under key "euler_number" within this dictionary.
+    pixels_per_thread : int, optional
+        Number of pixels processed per thread on the GPU during the final
+        weighted summation.
+
+    Returns
+    -------
+    euler_number : cp.ndarray of int
+        Euler characteristic of the set of all objects in the image.
+
+    Notes
+    -----
+    The Euler characteristic is an integer number that describes the
+    topology of the set of all objects in the input image. If object is
+    4-connected, then background is 8-connected, and conversely.
+
+    The computation of the Euler characteristic is based on an integral
+    geometry formula in discretized space. In practice, a neighborhood
+    configuration is constructed, and a LUT is applied for each
+    configuration. The coefficients used are the ones of Ohser et al.
+
+    It can be useful to compute the Euler characteristic for several
+    connectivities. A large relative difference between results
+    for different connectivities suggests that the image resolution
+    (with respect to the size of objects and holes) is too low.
+
+    References
+    ----------
+    .. [1] S. Rivollier. Analyse d’image geometrique et morphometrique par
+           diagrammes de forme et voisinages adaptatifs generaux. PhD thesis,
+           2010. Ecole Nationale Superieure des Mines de Saint-Etienne.
+           https://tel.archives-ouvertes.fr/tel-00560838
+    .. [2] Ohser J., Nagel W., Schladitz K. (2002) The Euler Number of
+           Discretized Sets - On the Choice of Adjacency in Homogeneous
+           Lattices. In: Mecke K., Stoyan D. (eds) Morphology of Condensed
+           Matter. Lecture Notes in Physics, vol 600. Springer, Berlin,
+           Heidelberg.
+    --------
+    perimeter_crofton
+
+    Examples
+    --------
+    >>> import cupy as cp
+    >>> from skimage import data
+    >>> from cucim.skimage import util
+    >>> from cucim.skimage.measure import label
+    >>> # coins image (binary)
+    >>> img_coins = cp.array(data.coins() > 110)
+    >>> # total perimeter of all objects in the image
+    >>> perimeter(img_coins, neighborhood=4)  # doctest: +ELLIPSIS
+    array(7796.86799644)
+    >>> perimeter(img_coins, neighborhood=8)  # doctest: +ELLIPSIS
```

Comment:
It looks like the examples section incorrectly shows `perimeter` function calls instead of `euler_number`.

Reply:
I should just remove these "see also" and examples as we are not exposing these versions directly as public API. It is just for internal use within `regionprops_table`